fontbe/src/gvar.rs

```diff
@@ -0,0 +1,101 @@
+//! Generates a [gvar](https://learn.microsoft.com/en-us/typography/opentype/spec/gvar) table.
+
+use std::cmp::{min, max};
+
+use font_types::{F2Dot14, GlyphId};
+use fontdrasil::orchestration::Work;
+use fontir::coords::NormalizedLocation;
+use write_fonts::{
+    dump_table,
+    tables::{
+        gvar::{GlyphDeltas, GlyphVariations, Gvar},
+        variations::Tuple,
+    },
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, GvarFragment},
+};
+
+struct GvarWork {}
+
+pub fn create_gvar_work() -> Box<BeWork> {
+    Box::new(GvarWork {})
+}
+
+fn to_tuple(loc: &NormalizedLocation) -> Tuple {
+    Tuple::new(
+        loc.iter()
+            .map(|(_, nc)| F2Dot14::from_f32(nc.to_f32()))
+            .collect(),
+    )
+}
+
+fn to_deltas(fragment: &GvarFragment) -> Vec<GlyphDeltas> {
+    fragment
+        .deltas
+        .iter()
+        .filter_map(|(loc, deltas)| {
+            if !loc.has_any_non_zero() {
+                return None;
+            }
+
+            // TODO: add phantoms properly
+            let mut min_x = i16::MAX;
+            let mut max_x = i16::MIN;
+            let mut min_y = i16::MAX;
+            let mut max_y = i16::MIN;
+            
+            
+            // TODO: nice rounding on deltas
+            let mut deltas: Vec<_> = deltas.iter().map(|v| {
+                let (x, y) = (v.x as i16, v.y as i16);
+                min_x = min(min_x, x);
+                max_x = max(max_x, x);
+                min_y = min(min_y, y);
+                max_y = max(max_y, y);                
+
+                (x, y)
+            }).collect();
+
+            // TODO: add phantoms properly
+            deltas.push((min_x, 0));
+            deltas.push((max_x, 0));
+            deltas.push((0, min_y));
+            deltas.push((0, max_y));
+
+            // TODO: what exactly is an intermediate region? Can we ignore it and let Gvar solve for it?
```

Comment:
> an intermediate region

it's the most general one, with three start/peak/end values; the special case with a single peak value is basically equivalent to start=0 and peak=end.
it's all described here: https://learn.microsoft.com/en-us/typography/opentype/spec/otvaroverview#variation-data

Reply:
Do we generally have intermediate regions, or are they often elided? I inferred from the docs that they are often elided, and I reflected that in the API, but maybe this was a mistake?

Reply:
of course, intermediate regions are very common. Think of a single-axis setup with two masters in addition to the default master: e.g. {Weight: 0.0, Weight: 0.5, Weight: 1.0}. You end up with the following intermediate regions:

```python
>>> model = VariationModel([{"Weight": 0.0}, {"Weight": 0.5}, {"Weight": 1.0}], ["Weight"])
>>> model.supports
[{}, {'Weight': (0, 0.5, 1.0)}, {'Weight': (0.5, 1.0, 1.0)}]
```

Only with very simple setups where you have only a master at the maximum ends of an axis, you get the simple (non-intermediate) sort:
```
>>> model = VariationModel([{"Weight": 0.0}, {"Weight": 1.0}], ["Weight"])
>>> model.supports
[{}, {'Weight': (0, 1.0, 1.0)}]
```

Reply:
Oh I see, I think I confused Tuple and Tent. I tend to think we will want to change the API to more explicitly model the tent structure.

Reply:
Updated.